### PR TITLE
Create cs_page.c

### PR DESCRIPTION
**Pull request by mzer0-yu**

### CODE REVIEW

File: core/pageProcesser/cs_page.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+int set_page(cs_page *p, char* context, unsigned int length);
+
+/*
+  clear_page: fill the data with all zeros.
+*/
+
+void clear_page(cs_page *p) {
+  if(p == NULL)
+    return;
+  if(p->data == NULL)
```
Comment (AntonioCS) [inline core/pageProcesser/cs_page.c:17]: This could be added to the line if condition above

File: core/pageProcesser/cs_page.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  if(capacity == 0)
+    return 0x10;
+  void* buf = malloc(capacity);
+  if(buf == NULL)
+    return 0x100;
+  p->data = buf;
+  p->capacity = capacity;
+  memset(p->data, 0, p->capacity);
+  /* set p->file_type and p->used to zeros */
+  memset(&p->file_type, 0, sizeof(cs_page) - &0->file_type);
```
Comment (AntonioCS) [inline core/pageProcesser/cs_page.c:52]: I believe &0->file_type should be &p->file_type

File: core/pageProcesser/cs_page.c [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+int set_page(cs_page *p, char* context, unsigned int length) {
+  if(p == NULL)
+    return 0x1;
+  if(context == NULL)
+    return 0x10;
+  if(p->length == 0)
+    return 0x100;
+  if(p->data == NULL) {
+    free(p->data);
+    goto label_next;
```
Comment (AntonioCS) [inline core/pageProcesser/cs_page.c:68]: Please use goto only for error management.

**Comment by luohaha** [comment]
Thanks for AntonioCS's correction!

File: core/pageProcesser/cs_page.c (hunk 2 above)
Comment (mzer0-yu) [inline core/pageProcesser/cs_page.c:52]: Absolutely not. &0->filye_type is a magic for calculating the length of 0 to file_type.

File: core/pageProcesser/cs_page.c (hunk 3 above)
Comment (mzer0-yu) [inline core/pageProcesser/cs_page.c:68]: My style.

File: core/pageProcesser/cs_page.c (hunk 2 above)
Comment (AntonioCS) [inline core/pageProcesser/cs_page.c:52]: And that is what you want to use? Magic?

File: core/pageProcesser/cs_page.c (hunk 3 above)
Comment (AntonioCS) [inline core/pageProcesser/cs_page.c:68]: That is not a valid reason to use goto outside of error management.

**Pull request merged**